lib/src/main/java/com/auth0/jwt/algorithms/RSAAlgorithm.java

```diff
@@ -70,14 +80,46 @@ public void verify(DecodedJWT jwt, EncodeType encodeType) throws Exception {
                 Base32 base32 = new Base32();
                 urlDecoded = URLDecoder.decode(signature, "UTF-8");
                 signatureBytes = base32.decode(urlDecoded);
+                System.out.println("signature bytes after being decoded: " + Arrays.toString(signatureBytes));
+                bytesAfterBeingDecoded = signatureBytes;
+                System.out.println("Are they equal? " + Arrays.equals(JWTCreator.bytesBeforeBeingDecoded, bytesAfterBeingDecoded));
                 break;
             case Base64:
                 signatureBytes = Base64.decodeBase64(signature);
+                System.out.println("signature bytes after being decoded: " + Arrays.toString(signatureBytes));
+                bytesAfterBeingDecoded = signatureBytes;
+                System.out.println("Are they equal? " + Arrays.equals(JWTCreator.bytesBeforeBeingDecoded, bytesAfterBeingDecoded));
                 break;
         }
 
         try {
-            RSAPublicKey publicKey = keyProvider.getPublicKeyById(jwt.getKeyId());
+            //RSAPublicKey publicKey = keyProvider.getPublicKeyById(jwt.getKeyId());
+            /*String kid = jwt.getKeyId();
+            JwkProvider provider = new UrlJwkProvider(new File("/Users/jdahmubed/documents/jwksRSA.json").toURI().toURL());
+            Jwk jwk = provider.get(kid);
+            PublicKey publicKey = jwk.getPublicKey();*/
+
+            String kid = jwt.getKeyId();
+            JwkProvider provider = new UrlJwkProvider(new File("/Users/jdahmubed/documents/jwksRSA.json").toURI().toURL());
+            Jwk jwk = provider.get(kid);
+            String cert = jwk.getCertificateChain().get(0);
+            try (Writer writer = new BufferedWriter(new OutputStreamWriter(
+                    new FileOutputStream("/Users/jdahmubed/workspace/javaOIDCMsgForCOMMENTS/lib/src/main/java/com/auth0/jwt/algorithms/jwks.cert"), "utf-8"))) {
```

Comment:
@jdahmubed paths shouldn't be hard coded for your machine

Reply:
right.  i can change that.